pkg/controllers/cloud/cloudprovideraccount_controller.go

```diff
@@ -215,25 +243,111 @@ func (r *CloudProviderAccountReconciler) startPollingThread(namespacedName *type
 
 func (r *CloudProviderAccountReconciler) addAccountProviderType(namespacedName *types.NamespacedName,
 	provider runtimev1alpha1.CloudProvider) common.ProviderType {
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
-
 	providerType := common.ProviderType(provider)
 	r.accountProviderType[*namespacedName] = providerType
-
 	return providerType
 }
 
 func (r *CloudProviderAccountReconciler) removeAccountProviderType(namespacedName *types.NamespacedName) {
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
 
 	delete(r.accountProviderType, *namespacedName)
 }
 
 func (r *CloudProviderAccountReconciler) getAccountProviderType(namespacedName *types.NamespacedName) common.ProviderType {
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
 
 	return r.accountProviderType[*namespacedName]
 }
+
+// getCpaBySecret returns nil only when the Secret is not used by any CloudProvideAccount CR,
+// otherwise the dependent CloudProvideAccount CR will be returned.
+func (r *CloudProviderAccountReconciler) getCpaBySecret(s types.NamespacedName) (error, []crdv1alpha1.CloudProviderAccount) {
+	cpaList := &crdv1alpha1.CloudProviderAccountList{}
+	if err := r.Client.List(context.TODO(), cpaList, &client.ListOptions{}); err != nil {
+		return fmt.Errorf("failed to get CloudProviderAccount list, err:%v", err), nil
```

Comment:
nit: for consistency
```suggestion
		return fmt.Errorf("failed to get CloudProviderAccount list, err %v", err), nil
```

Reply:
Solved.